### PR TITLE
Issue 810

### PR DESCRIPTION
**Pull request by deepakaravindr**
2 commits
1. Clean-up commit. Fixed code to follow coding guidelines
2. Fixes NuGet/Home#810
   Implements an ILogger on OnBuildPackageRestorer and
   update the progress dialog, write to output window and add
   an error/warning to error list window as appropriate

@yishaigalatzer , @emgarten, @pranavkm

### CODE REVIEW

**Comment by emgarten** [comment]
Sign your cla already :smiley:

**Comment by yishaigalatzer** [comment]
:watch: on the perf concerns, it feels like this can slow down things significantly. Do we need to switch to the main thread for both the UI and the output window? Can we see in what mode we need to write messages so we can avoid the more verbose calls? Can we split and write say a message a second into the window and all of them into the output window?

**Comment by deepakaravindr** [comment]
I don't find the restore experience to be very slow. It is fairly smooth. @yishaigalatzer @emgarten

Having said that,

Need to Switch to the main thread to write to the output window and error list window. Also, that cannot slow things down as much as we think. Since the log methods on ILogger are synchronous, we will have to block the calling worker thread using JTF.Run and switch to the main thread. That can slightly slow things down. Originally, I had JTF.RunAsync instead of JTF.Run. This is a fire and forget and can make the experience smoother. But, it is possible that the messages do not show up in order and based on comment from Andrew Arnott, I changed it to JTF.Run. And, IMO, the experience is still pretty good

**Comment by emgarten** [comment]
This looks good to me. We'll need to wait for approval to check it in.

**Comment by deepakaravindr** [comment]
@yishaigalatzer  @pranavkm... Ping!

File: src/VsExtension/VsEvents/OnBuildPackageRestorer.cs [hunk 1]
```diff
@@ -598,5 +594,86 @@ public void Dispose()
             _buildEvents.OnBuildBegin -= BuildEvents_OnBuildBegin;
             _solutionEvents.AfterClosing -= SolutionEvents_AfterClosing;
         }
+
+        #region ILogger implementation
+        public void LogDebug(string data)
```
Comment (pranavkm) [inline src/VsExtension/VsEvents/OnBuildPackageRestorer.cs:599]: Would it make sense to have a separate logger type? There's no reason to tie it to the restorer is there?

**Comment by pranavkm** [comment]
Spoke to @deepakaravindr about the refactoring - this could be done as a follow up to avoid churn at this point.

:shipit:

File: src/VsExtension/VsEvents/OnBuildPackageRestorer.cs (hunk 1 above)
Comment (yishaigalatzer) [inline src/VsExtension/VsEvents/OnBuildPackageRestorer.cs:599]: agree

File: src/VsExtension/VsEvents/OnBuildPackageRestorer.cs [hunk 2]
```diff
@@ -598,5 +594,86 @@ public void Dispose()
...
+
+        public void LogError(string data)
+        {
+            LogToVS(VerbosityLevel.Quiet, data);
+        }
+        #endregion ILogger implementation
+
+        private void LogToVS(VerbosityLevel verbosityLevel, string message)
+        {
+            if (Token.IsCancellationRequested)
```
Comment (yishaigalatzer) [inline src/VsExtension/VsEvents/OnBuildPackageRestorer.cs:627]: add a comment explaining why you are checking for cancellation inside a logger (it is very weird), also setting the canceled flag in here is odd and will not stand when you pull the logger out

**Pull request closed without merging**